src/components/Filters/index.tsx

```diff
@@ -85,14 +85,20 @@ const Filters = ({
 
 	// unfortunately we have to keep all these useEffects in order
 	// to synchronize the state of this component with that of the parent
-	React.useEffect(
-		() =>
-			onFiltersUpdate &&
-			onFiltersUpdate(
-				filtersState.map(filter => SchemaSieve.unflattenSchema(filter)),
-			),
-		[filtersState],
-	);
+	React.useEffect(() => updateParentFilters(filtersState), [filtersState]);
+
+	React.useEffect(() => {
+		const newFilters = searchString
+			? SchemaSieve.upsertFullTextSearch(
+					schemaState,
+					filtersState,
+					searchString,
+			  )
+			: SchemaSieve.removeFullTextSearch(filtersState);
+
+		setFiltersState(newFilters);
```

Comment:
If this changes the filtersState, then the effect above should fire. Like this the parent callback will be called twice, which is not really desired.

Reply:
I just tried and it doesn't fire, but I don't get it why. This whole component needs some refactoring now that it became even more confusing. Can you please look into simplifying this once the bug is fixed? Thanks